src/transformers/models/mplugdocowl/image_processing_mplugdocowl.py

```diff
@@ -0,0 +1,707 @@
+    "grid_9": [
+        (1, 1),
+        (1, 2),
+        (2, 1),
+        (1, 3),
+        (3, 1),
+        (2, 2),
+        (1, 4),
+        (4, 1),
+        (1, 5),
+        (5, 1),
+        (1, 6),
+        (6, 1),
+        (2, 3),
+        (3, 2),
+        (1, 7),
+        (7, 1),
+        (4, 2),
+        (2, 4),
+        (1, 8),
+        (8, 1),
+        (3, 3),
+        (1, 9),
+        (9, 1),
+    ],
+    "grid_3x3": [(3, 3)],
+    "grid_20": [
+        (1, 1),
+        (1, 2),
+        (2, 1),
+        (1, 3),
+        (3, 1),
+        (1, 4),
+        (2, 2),
+        (4, 1),
+        (1, 5),
+        (5, 1),
+        (1, 6),
+        (2, 3),
+        (3, 2),
+        (6, 1),
+        (1, 7),
+        (7, 1),
+        (1, 8),
+        (2, 4),
+        (4, 2),
+        (8, 1),
+        (1, 9),
+        (3, 3),
+        (9, 1),
+        (1, 10),
+        (2, 5),
+        (5, 2),
+        (10, 1),
+        (1, 11),
+        (11, 1),
+        (2, 6),
+        (3, 4),
+        (4, 3),
+        (6, 2),
+        (2, 7),
+        (7, 2),
+        (3, 5),
+        (5, 3),
+        (2, 8),
+        (4, 4),
+        (8, 2),
+        (2, 9),
+        (3, 6),
+        (6, 3),
+        (9, 2),
+        (2, 10),
+        (4, 5),
+        (5, 4),
+        (10, 2),
+    ],
+}
+
+
+def box_area(boxes):
+    r"""
+    Compute the area of each bounding box in a given set of bounding boxes.
+
+    Args:
+        boxes (np.ndarray): An array of shape (N, 4) containing N bounding boxes,
+                            each represented by the coordinates [x_min, y_min, x_max, y_max].
+
+    Returns:
+        np.ndarray: An array of shape (N,) containing the area of each bounding box.
+    """
+    return (boxes[:, 2] - boxes[:, 0]) * (boxes[:, 3] - boxes[:, 1])
+
+
+def box_iou(boxes1, area1, boxes2, eps=1e-5):
+    r"""
+    Compute the Intersection over Union (IoU) between two sets of bounding boxes.
+
+    Args:
+        boxes1 (np.ndarray): An array of shape (N, 4) containing N bounding boxes.
+        area1 (np.ndarray): An array of shape (N,) containing the area of each bounding box in boxes1.
+        boxes2 (np.ndarray): An array of shape (M, 4) containing M bounding boxes.
+        eps (float, optional): A small value to avoid division by zero. Defaults to 1e-5.
+
+    Returns:
+        tuple: A tuple containing:
+            - np.ndarray: An array of shape (N, M) containing the IoU between each pair of boxes from boxes1 and boxes2.
+            - np.ndarray: An array of shape (N, M) containing the union areas of each pair of boxes.
+    """
+    area2 = box_area(boxes2)
+
+    lt = np.maximum(boxes1[:, None, :2], boxes2[:, :2])  # [N,M,2]
+    rb = np.minimum(boxes1[:, None, 2:], boxes2[:, 2:])  # [N,M,2]
+
+    wh = np.clip(rb - lt, a_min=0, a_max=None)  # [N,M,2]
+    inter = wh[:, :, 0] * wh[:, :, 1]  # [N,M]
+
+    union = area1[:, None] + area2 - inter
+
+    iou = inter / (union + eps)
+
+    return iou, union
```

Comment:
Sure:

`def box_iou(boxes1, area1, boxes2, eps=1e-5):
    r"""
    Compute the Intersection over Union (IoU) between two sets of bounding boxes.

    Args:
        boxes1 (np.ndarray): An array of shape (N, 4) containing N bounding boxes.
        area1 (np.ndarray): An array of shape (N,) containing the area of each bounding box in boxes1.
        boxes2 (np.ndarray): An array of shape (M, 4) containing M bounding boxes.
        eps (float, optional): A small value to avoid division by zero. Defaults to 1e-5.

    Returns:
        tuple: A tuple containing:
            - np.ndarray: An array of shape (N, M) containing the IoU between each pair of boxes from boxes1 and boxes2.
            - np.ndarray: An array of shape (N, M) containing the union areas of each pair of boxes.
    """
    area2 = box_area(boxes2)

    top_left = np.maximum(boxes1[:, None, :2], boxes2[:, :2])  # [N,M,2]
    bottom_right = np.minimum(boxes1[:, None, 2:], boxes2[:, 2:])  # [N,M,2]

    width_height = np.clip(bottom_right - top_left, a_min=0, a_max=None)  # [N,M,2]
    intersection = width_height[:, :, 0] * width_height[:, :, 1]  # [N,M]

    union = area1[:, None] + area2 - intersection

    iou = intersection / (union + eps)

    return iou, union`